### PR TITLE
Audit log improvements

### PR DESCRIPTION
**Pull request by chathurabuddi**
EXTGW-618

- Tier-edit audit log enhancements. 
- Application approval audit logs cherry-picked & squashed from INTGW-1199
- Subscription approval audit logs cherry-picked & squashed from INTGW-1195

Following audit logs are added.

_Application Creation_
```
[2020-12-18 01:50:08,200]  INFO -  Application creation approval workflow submitted. | Workflow ID: 5f018077-3fea-4359-89cf-193029c7e274 | Workflow Status: CREATED | Application Name: apigate-app-2 | Application ID: 0 | Subscriber: apigate-sp | Requested Tier: Default
```

_Application Approval_
```
[2020-12-18 01:51:15,692]  INFO -  Application creation approval process completed. | Workflow ID: 5f018077-3fea-4359-89cf-193029c7e274 | Workflow State: APPROVED | Application Name: apigate-app-2 | Application ID: 7 | Subscriber: apigate-sp | Requested Tier: Default | Approved Tier: Default | Performed By: admin
```

_Subscription Creation_
```
[2020-12-18 02:03:20,182]  INFO -  Subscription creation approval workflow submitted. | Workflow ID: b0a1c93f-c14e-4081-8cb1-f79bdef396cd | Workflow Status: CREATED | API: payment:v1 | Application: apigate-app-2 | Application ID: 0 | Subscriber: apigate-sp | Requested Tier: Default
```

_Subscription Approval_
```
[2020-12-18 02:04:13,147]  INFO -  Subscription creation approval workflow completed. | Workflow ID: b0a1c93f-c14e-4081-8cb1-f79bdef396cd | Workflow State: APPROVED | API: payment:v1 | Application: apigate-app-2 | Application ID: 7 | Subscriber: apigate-sp | Requested Tier: Default | Approved Tier: Default | Performed By: admin
```

_Application Tier Edit_
```
[2020-12-18 03:20:01,081]  INFO -  Application Updated. | Application: 2:apigate-app | SP: apigate-sp | Previous Tier: Unlimited | Updated Tier: Default | User: admin
```

_Subscription Tier Edit_
```
[2020-12-18 02:04:46,491]  INFO -  Subscription Updated. | Application: 2:apigate-app | API: 1:payment:v1 | Previous Tier: Default | Updated Tier: Unlimited | User: admin
```

### CODE REVIEW

File: components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/ApplicationCreationRestWorkflowExecutor.java [hunk 1]
```diff
@@ -307,9 +317,19 @@ public void cleanUpPendingTask(String workflowExtRef) throws WorkflowException {
...
+        log.info(logm);
+        auditLog.info(logm);
 
+        try {
+            Application app = dao.getApplicationById(Integer.parseInt(applicationId));
+            String audit = "{\"action\":\"deleted\",\"typ\":\"Application\",\"info\":\"{\\\"tier\\\":"+app.getTier()+",\\\"name\\\":"+app.getName()+",\\\"callbackURL\\\":"+app.getCallbackUrl()+"}\"}";
+            log.info(audit);
+
+        } catch (APIManagementException e) {
+            e.printStackTrace();
```
Comment (aushaniU) [inline components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/ApplicationCreationRestWorkflowExecutor.java]: avoid printStackTrace()

File: components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java [hunk 2]
```diff
@@ -68,6 +68,28 @@ public Callback searchPending(TaskSearchDTO searchDTO, UserProfileDTO userProfil
         return buildMyTaskResponse(searchDTO, taskList, userProfile);
     }
 
+    @Override
+    public Callback searchPendingById(String taskId, final UserProfileDTO userProfile, String workflowType) throws BusinessException {
```
Comment (aushaniU) [inline components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java:72]: add javadoc comments for method and logs to identify the purpose of the . Apply this change for other new methods too

File: components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java [hunk 3]
```diff
@@ -68,6 +68,28 @@ public Callback searchPending(TaskSearchDTO searchDTO, UserProfileDTO userProfil
...
+            if (APIUtil.isAdvanceThrottlingEnabled() && "APPLICATION".equals(workflowType)) {
+                String apiTiers = getTiersFromDB(workflowType);
+                TaskVariableResponse[] vars = activityClient.getVariables(taskId);
+                task.setVariables(replaceiActivitiTiers(vars, apiTiers, workflowType));
+            } else {
+                TaskVariableResponse[] vars = activityClient.getVariables(taskId);
+                task.setVariables(vars);
+            }
+        } catch (WorkflowExtensionException e) {
+            log.error(e.getMessage(), e);
```
Comment (aushaniU) [inline components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java]: No need to log and re-throw again. Hence can remove log message here

File: components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/SubscriptionCreationRestWorkflowExecutor.java [hunk 4]
```diff
@@ -314,12 +328,18 @@ public WorkflowResponse complete(WorkflowDTO workFlowDTO) throws WorkflowExcepti
                 try {
                     apiMgtDAO.updateSubscriptionStatus(Integer.parseInt(workFlowDTO.getWorkflowReference()), status);
                 } catch (APIManagementException e) {
-                    log.error("Could not complete subscription creation workflow", e);
-                    throw new WorkflowException("Could not complete subscription creation workflow", e);
+                    String errorMsg = "Could not complete subscription approval workflow." +
+                            " | Workflow ID: " + workFlowDTO.getExternalWorkflowReference();
+                    log.error(errorMsg, e);
```
Comment (aushaniU) [inline components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/SubscriptionCreationRestWorkflowExecutor.java]: Can remove this log error

File: components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/ApplicationCreationRestWorkflowExecutor.java (hunk 1 above)
Comment (chathurabuddi) [reply components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/ApplicationCreationRestWorkflowExecutor.java]: This is fixed. replaced with a log line

File: components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java (hunk 3 above)
Comment (chathurabuddi) [reply components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java]: Fixed. logline removed

File: components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/SubscriptionCreationRestWorkflowExecutor.java (hunk 4 above)
Comment (chathurabuddi) [reply components/workflow-extensions/src/main/java/com/wso2telco/hub/workflow/extensions/SubscriptionCreationRestWorkflowExecutor.java]: removed

File: components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java (hunk 2 above)
Comment (chathurabuddi) [reply components/workflow-core/src/main/java/org/workflow/core/service/AbsractQueryBuilder.java:72]: It's better not to add Javadoc comments unless it's an external API method to maintain a shorter and clearer code.

**Review by aushaniU — APPROVED** [review]

**Pull request merged**